packages/strapi-admin/admin/src/containers/Webhooks/ListView/index.js

```diff
@@ -0,0 +1,273 @@
+    webhooks.findIndex(webhook => webhook.id === id);
+
+  // New button
+  const addBtnLabel = formatMessage({
+    id: `Settings.webhooks.list.button.add`,
+  });
+
+  const newButtonProps = {
+    label: addBtnLabel,
+    onClick: () => handleGoTo('create'),
+    color: 'primary',
+    type: 'button',
+    icon: <Plus fill="#007eff" width="11px" height="11px" />,
+  };
+
+  // Header props
+  const actions = [
+    {
+      ...newButtonProps,
+      icon: true,
+      style: {
+        paddingLeft: 15,
+        paddingRight: 15,
+      },
+    },
+  ];
+
+  const headerProps = {
+    title: {
+      label: formatMessage({ id: `Settings.webhooks.title` }),
+    },
+    content: formatMessage({ id: `Settings.webhooks.list.description` }),
+    actions: actions,
+  };
+
+  // List props
+  const rowsCount = webhooks.length;
+  const titleLabel = `${
+    rowsCount > 1
+      ? formatMessage({ id: `Settings.webhooks.title` })
+      : formatMessage({ id: `Settings.webhooks.singular` })
+  }`;
+  const title = `${rowsCount} ${titleLabel}`;
+
+  const buttonProps = {
+    color: 'delete',
+    disabled: webhooksToDelete.length > 0 ? false : true,
+    label: formatMessage({ id: `Settings.webhooks.list.button.delete` }),
+    onClick: () => setShowModal(true),
+    type: 'button',
+  };
+
+  const listProps = {
+    title,
+    button: buttonProps,
+    items: webhooks,
+  };
+
+  const fetchData = async () => {
+    try {
+      const { data } = await request(`/admin/webhooks`, {
+        method: 'GET',
+      });
+
+      if (isMounted.current) {
+        dispatch({
+          type: 'GET_DATA_SUCCEEDED',
+          data,
+        });
+      }
+    } catch (err) {
+      if (isMounted.current) {
+        if (err.code !== 20) {
+          strapi.notification.error('notification.error');
+        }
+      }
+    }
+  };
+
+  const handleChange = (value, id) => {
+    dispatch({
+      type: 'SET_WEBHOOKS_TO_DELETE',
+      value,
+      id,
+    });
+  };
+
+  const handleConfirmDelete = () => {
+    if (webhookToDelete) {
+      handleConfirmDeleteOne();
+    } else {
+      handleConfirmDeleteAll();
+    }
+  };
+
+  const handleConfirmDeleteOne = async () => {
+    try {
+      await request(`/admin/webhooks/${webhookToDelete}`, {
+        method: 'DELETE',
+      });
+
+      dispatch({
+        type: 'WEBHOOK_DELETED',
+        index: getWebhookIndex(webhookToDelete),
+      });
+    } catch (err) {
+      if (err.code !== 20) {
+        strapi.notification.error('notification.error');
+      }
+    }
+    setShowModal(false);
+  };
+
+  const handleConfirmDeleteAll = async () => {
+    const body = {
+      ids: webhooksToDelete,
+    };
+
+    try {
+      await request(`/admin/webhooks/batch-delete`, {
+        method: 'POST',
+        body,
+      });
+
+      if (isMounted.current) {
+        dispatch({
+          type: 'WEBHOOKS_DELETED',
+        });
+      }
+    } catch (err) {
+      if (isMounted.current) {
```

Comment:
Not sure this is needed here but the `setShowModal(false)` should be inside the `isMounted` condition